stratum/hal/lib/barefoot/bf_sde_wrapper.cc

```diff
@@ -567,15 +567,63 @@ ::util::Status TableData::SetActionMemberId(uint64 action_member_id) {
 }
 
 ::util::Status TableData::GetActionMemberId(uint64* action_member_id) const {
-  return GetField(*table_data_, "$ACTION_MEMBER_ID", action_member_id);
+  const bfrt::BfRtTable* table;
+  RETURN_IF_BFRT_ERROR(table_data_->getParent(&table));
+  // Here we assume that table entries with action IDs (direct match-action) can
+  // never hold action member or group IDs (indirect match-action). Since this
+  // function is regularly called on both, we do not log this error here.
+  if (table->actionIdApplicable()) {
+    return MAKE_ERROR(ERR_ENTRY_NOT_FOUND).without_logging()
+           << "Table data does not contain an action member ID.";
```

Comment:
Since we are checking the "table" in the `if` condition, maybe it can be something like "Table does not contain an action member ID field."? same comment for the action group one